pkg/controllers/clustermanagementaddons/managedclusterinfo.go

```diff
@@ -0,0 +1,263 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package clustermanagementaddons
+
+import (
+	"context"
+	"reflect"
+
+	clusterinfov1beta1 "github.com/stolostron/cluster-lifecycle-api/clusterinfo/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/stolostron/multicluster-controlplane/pkg/controllers/utils"
+)
+
+const (
+	clusterFinalizerName = "managedclusterinfo.finalizers.open-cluster-management.io"
+)
+
+var (
+	logCertSecretNamespace string
+	logCertSecretName      string
+)
+
+type ClusterInfReconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func SetupClusterInfoWithManager(mgr manager.Manager, kubeClient kubernetes.Interface) error {
```

Comment:
Ideally, we should call the controller in https://github.com/stolostron/multicloud-operators-foundation/tree/main/pkg/controllers/clusterinfo. It may have go dependency conflicts in open-cluster-management.io. But it is worth to try in stolostron.

Reply:
@clyang82 If we call the controller from https://github.com/stolostron/multicloud-operators-foundation/tree/main/pkg/controllers/clusterinfo. Then we have to create a `logCertSecret` from the second parameter.
https://github.com/stolostron/multicloud-operators-foundation/blob/f8043f62d6f77dcea430b83986e7bba8f24f1821/pkg/controllers/clusterinfo/clusterinfo_controller.go#L46